make-util.js

```diff
@@ -154,10 +156,20 @@ var buildNodeTask = function (taskPath, outDir) {
     var originalDir = pwd();
     cd(taskPath);
     var packageJsonPath = rp('package.json');
+    var tscPath;
     if (test('-f', packageJsonPath)) {
         // verify no dev dependencies
+        // we allow a TS dev-dependency to indicate a task should use a different TS version
         var packageJson = JSON.parse(fs.readFileSync(packageJsonPath).toString());
-        if (packageJson.devDependencies && Object.keys(packageJson.devDependencies).length != 0) {
+        var devDeps = packageJson.devDependencies ? Object.keys(packageJson.devDependencies).length != 0 : 0;
+        if (devDeps == 1 && packageJson.devDependencies["typescript"]) {
+            var version = packageJson.devDependencies["typescript"];
+            if (!allowedTypescriptVersions.includes(version)) {
+                fail('The package.json specifies a different TS version that the allowed versions: ' + allowedTypescriptVersions + '. Offending package.json: ' + packageJsonPath);
+            }
+            tscPath = path.join(taskPath, "node_modules", "typescript");
+            console.log(`Detected Typescript version: ${version}`);
+        } else if (devDeps > 1) {
```

Comment:
I think this needs to be `>=` right? For the case where there is 1 dev dependency that isn't TS.

Reply:
Yes! Good catch